lua/entities/magicbutton.lua

```diff
@@ -0,0 +1,463 @@
+﻿AddCSLuaFile()
+ENT.Type = "anim"
+ENT.PrintName = "Secret Button"
+ENT.Author = "PYROTEKNIK"
+ENT.Purpose = "If you find one of these, something magic may happen"
+ENT.Category = "PYROTEKNIK"
+ENT.Spawnable = true
+ENT.AdminSpawnable = true
+ENT.PlacementSettings = {}
+ENT.PlacementSettings.RequireCeiling = false
+ENT.PlacementSettings.AllowFloor = false
+--NOTE: Generally the process will be less efficient if both of these are set to false!
+ENT.PlacementSettings.CeilingHeight = 400
+
+ENT.PlacementSettings.SurfacePropBlacklist = {"dirt", "grass", "gravel", "rock"}
+
+--**studio** is any model
```

Comment:
If models don't work, what about displacements? I think there is a special material for them too

Reply:
some displacements are not terrain. These checks are intended to prevent placing them on surfaces that don't make sense for a button to be, like floating on a disjointed part of a model's collision geometry, or on natural surfaces.